custom/icds_reports/management/commands/dashboard_usage_report.py

```diff
@@ -134,7 +134,7 @@ def get_tabular_data(self, usage_data, start_date, end_date, domain):
     def get_cas_data(self, username_state_mapping, start_date, end_date, domain):
         cas_total_counts = self.get_dashboard_cas_usage_counts(start_date, end_date, domain)
         sheet_headers = ['Sr.No', 'State/UT Name',
-                         'No. of times CAS data export downloaded (December 2019)']
+                         f'No. of times CAS data export downloaded ({start_date.strftime("%B %Y")})']
```

Comment:
What is `end_date` in this context? Can start_date -> end_date span more than 1 month?

Reply:
The command is meant to generate monthly report for dashboard_usage so the start_date->end_date is never more than 1 month and end_date is the last date of that month for which we are pulling data.

Reply:
Along with this fix could you fix the end_date filter here: https://github.com/dimagi/commcare-hq/pull/27354/files#diff-4f218f2e38f9e93d34c15bc7a4fe5699R78 and on the other place as well.
By that I mean to change `time_of_use__lte`  to `time_of_use__lt` This will help us handle the corner of boundary submission elegantly

Reply:
could you also add a check here:https://github.com/dimagi/commcare-hq/pull/27354/files#diff-4f218f2e38f9e93d34c15bc7a4fe5699R143
`if key in username_state_mapping`

Reply:
updated

Reply:
This change looks better.

Relatedly, if the end date is fixed (and only valid if exactly 1 month after the start date), it probably doesn't make sense as an argument for the command. Right now someone can put anything as the end date and the command will run with that. There is nothing about it being monthly

Reply:
updated added a range(start date to end date) instead of start date or end date as discussed